examples/00_basic/server.py

```diff
@@ -1,102 +1,172 @@
+app.config['SQLALCHEMY _DATABASE_URI'] = 'something'
+
+db = SQLAlchemy(app)
+blacklist = []
+class User(db.Model):
+    ''' Setting up User'''
+    id = db.Column(db.Integer, primary_key=True)
+    # User vereification
+    email = db.Column(db.String(300), nullable=False, unique=True)
+    username = db.Column(db.String(60), nullable=False, unique=True)
+    password = db.Column(db.String(300), nullable=False)
+    active = db.Column(db.String(50),nullable=False)
+    token = db.Column(db.String(500),nullable = True)
+    refrsh_token = db.Column(db.String(500),nullable = True)
+    # User details
+    name = db.Column(db.String(50), nullable=False)
+
+
+def encoding_password(password):
+    return sc.hash(password])
+
+#check if the invalid token has expired or not
+def invalidate_token():
+    for invalid in blacklist:
+        try:
+            jwt.verify(invalid,app.config['SECRET_KEY'])
+        except:
+            blacklist.remove(invalid)
+#validating refresh token
+def validate_ref_token(token):
+    try:
+        jwt.verify(token,secret_key_2)
+        return True
+    except:
+        return False
+#generating refresh token
+def generate_refresh_token(user):
+    tokn= jwt.encode(
+            {
+                "id":user.id,
+                "iat": datetime.datetime.utcnow(),
+                "exp": datetime.datetime.utcnow() + datetime.timedelta(hours=2)
+            },
+            secret_key_2
+        )
+    return tokn
+#generate access token
+def generate_access_token(user):
+     tokn= jwt.encode(
+            {
+                "id":user.id,
+                "iat": datetime.datetime.utcnow(),
+                "exp": datetime.datetime.utcnow() + datetime.timedelta(minutes=2)
+            },
+            secret_key_2
+        )
+    return tokn
+
+#creating decorator
+#can be used for email auth
+def require_login(func):
+    @wraps(func)
+    def checking(*args,**kwargs):
+        token = request.args.get('token')
+        ref_token = request.args.get('r_token')
+        invalidate_token()
+        if not token or token in blacklist:
+            return jsonify({'message':  'Missing token'}),403
+        try:
+            data = jwt.decode(token, app.config['SECRET_KEY'])
+            verified_user = User.query.filter_by(id = data['id']).first()
+            refresh_case = False
+        except:
+            try:
+                exist_data = jwt.decode(ref_token,secret_key_2)
+                verified_user = User.query.filter_by(id = data['id']).first()
+                refresh_case = True
+            except:
+                return jsonify({'message': 'token is invalid'}),403
+        return func(verified_user,refresh_case,*args, **kwargs)
+    return checking
+
+@app.route('/',methods=['GET','POST'])
+def index():
+    return 'Welcome to chatbox'
+
+#go to user home
+@app.route('/home/<token><r_token>',methods=['GET','POST'])
+@require_login
+def user_home(current_user,ref_case):
+    if current_user.active == "True":
```

Comment:
No variable called current_user

Reply:
In here the decorator is returning verified user to current user